src/charm.py

```diff
@@ -439,25 +443,40 @@ def k8s_auth(self):
 
         self._authed = True
 
+    def k8s_api(self, api: str) -> object:
+        """Authenticate and return a k8s API.
+
+        Args:
+            api: Which API should be returned. If is not found, None is returned.
+        """
+        self.k8s_auth()
+        if api == KUBERNETES_CORE_API:
+            return _core_v1_api()
+        if api == KUBERNETES_NETWORKING_API:
+            return _networking_v1_api()
+        LOGGER.error("k8s api %s not found", api)
+        return None
+
     def _report_service_ips(self) -> List[str]:
         """Report on service IP(s) and return a list of them."""
-        self.k8s_auth()
-        api = _core_v1_api()
-        services = api.list_namespaced_service(namespace=self._namespace)
+        api = self.k8s_api(KUBERNETES_CORE_API)
```

Comment:
Is this an improvement? Maybe we have a function `core_v1_api` that returns the `core_v1_api` after calling `k8s_auth`? We probably lose a lot of type hint value by making the api to use generic

Reply:
I might misunderstand one of your reviews when you said "Take the API is a parameter?" so I created this method receiving the wanted API as a parameter. I'll review if it's ok to change every call to core_v1_api/networking to include self.k8s_auth instead.